### ISSUE TITLE
Compile error on Windows (Visual Studio 2017) with in pattern_formatter-inl.h (with std::chrono)

### ISSUE THREAD

**Issue by bsergean** [gabime/spdlog#1251]
I started getting that error after upgrading to latest master (as of today or tomorrow), and compiling the lib **by using cmake include_directory** instead of using a single header file.

```
c:\projects\ixwebsocket\third_party\spdlog\include\spdlog\details\pattern_formatter-inl.h(881): error C2589: '(': illegal token on right side of '::'
c:\projects\ixwebsocket\third_party\spdlog\include\spdlog\details\pattern_formatter-inl.h(880): note: while compiling class template member function 'void spdlog::details::elapsed_formatter<Padder,std::chrono::seconds>::format(const spdlog::details::log_msg &,const tm &,spdlog::memory_buf_t &)'
        with
        [
            Padder=spdlog::details::null_scoped_padder
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\type_traits(616): note: see reference to class template instantiation 'spdlog::details::elapsed_formatter<Padder,std::chrono::seconds>' being compiled
        with
        [
            Padder=spdlog::details::null_scoped_padder
        ]
c:\projects\ixwebsocket\third_party\spdlog\include\spdlog\details\pattern_formatter-inl.h(1216): note: see reference to class template instantiation 'std::is_convertible<spdlog::details::elapsed_formatter<Padder,std::chrono::seconds> *,_Ty *>' being compiled
        with
        [
            Padder=spdlog::details::null_scoped_padder,
            _Ty=spdlog::details::flag_formatter
        ]
c:\projects\ixwebsocket\third_party\spdlog\include\spdlog\details\pattern_formatter-inl.h(881): note: see reference to variable template 'const bool conjunction_v<std::negation<std::is_array<spdlog::details::elapsed_formatter<spdlog::details::null_scoped_padder,std::chrono::duration<__int64,std::ratio<1,1> > > > >,std::is_convertible<spdlog::details::elapsed_formatter<spdlog::details::null_scoped_padder,std::chrono::duration<__int64,std::ratio<1,1> > > *,spdlog::details::flag_formatter *>,std::is_convertible<std::default_delete<spdlog::details::elapsed_formatter<spdlog::details::null_scoped_padder,std::chrono::duration<__int64,std::ratio<1,1> > > >,std::default_delete<spdlog::details::flag_formatter> > >' being compiled
c:\projects\ixwebsocket\third_party\spdlog\include\spdlog\details\pattern_formatter-inl.h(1295): note: see reference to function template instantiation 'void spdlog::pattern_formatter::handle_flag_<spdlog::details::null_scoped_padder>(char,spdlog::details::padding_info)' being compiled
c:\projects\ixwebsocket\third_party\spdlog\include\spdlog\details\pattern_formatter-inl.h(881): error C2062: type 'unknown-type' unexpected
c:\projects\ixwebsocket\third_party\spdlog\include\spdlog\details\pattern_formatter-inl.h(881): error C2059: syntax error: ')'
c:\projects\ixwebsocket\third_party\spdlog\include\spdlog\details\pattern_formatter-inl.h(882): error C3536: 'delta': cannot be used before it is initialized
c:\projects\ixwebsocket\third_party\spdlog\include\spdlog\details\pattern_formatter-inl.h(882): error C2672: 'std::chrono::duration_cast': no matching overloaded function found
c:\projects\ixwebsocket\third_party\spdlog\include\spdlog\details\pattern_formatter-inl.h(882): error C2784: '_To std::chrono::duration_cast(const std::chrono::duration<_Rep,_Period> &)': could not deduce template argument for 'const std::chrono::duration<_Rep,_Period> &' from 'int'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\chrono(64): note: see declaration of 'std::chrono::duration_cast'
c:\projects\ixwebsocket\third_party\spdlog\include\spdlog\details\pattern_formatter-inl.h(885): error C3536: 'delta_units': cannot be used before it is initialized
c:\projects\ixwebsocket\third_party\spdlog\include\spdlog\details\pattern_formatter-inl.h(885): error C2672: 'spdlog::details::fmt_helper::pad6': no matching overloaded function found
```

The code is run on CI through appveyor ; it is a CMake build.
https://github.com/machinezone/IXWebSocket/blob/master/appveyor.yml

Any help will be appreciated :)

**Comment by bsergean** [comment]
This looks like the same error as https://github.com/gabime/spdlog/issues/902
and indeed I'm using some std::min and std::max in my code.

**Comment by bsergean** [comment]
The full error is here -> https://ci.appveyor.com/project/bsergean/ixwebsocket/builds/27756639
Hope this is accessible.

**Comment by bsergean** [comment]
Problem with those 3 lines maybe ? I'll try to use the same fix as in #902 and see if it helps.

```
include/spdlog/fmt/bundled/printf.h
47:    return (std::max)(static_cast<int>(value), 0);

include/spdlog/fmt/bundled/format.h
1041:    int num_zeros = (std::max)(params.num_digits - full_exp, 1);

include/spdlog/details/pattern_formatter-inl.h
881:        auto delta = std::max(msg.time - last_message_time_, log_clock::duration::zero());

```

**Comment by gabime** [comment]
Fixed in #1252
